### PR TITLE
refactor: handle non-stacks accounts 

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5671144193).

Seeing what's involved in remove dependence on Stacks keys.

To test this, I made a wallet with all the keys, then removed the Stacks keys. This breaks the wallet. To fix this, I had to start decomposing hooks and components to structure them conditionally. 

Notice how previously we had a component that relies on the current stacks account from a hook.

```ts
export function useAccountNonFungibleTokenHoldings() {
  const currentAccount = useCurrentStacksAccount();
```

This approach doesn't work when keys might be `undefined`. Instead, can create components that return the data we need, or not, so that we have inner components that consume the data as though it's always defined. See `CurrentStacksAccountLoader`.

Much similar refactoring is needed to further remove reliance on the Stacks features. As a rule of thumb, I suggest that from now on we favour passing in data into hooks at the component level, rather than assuming "current" data and having hooks rely on this.

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]
🚀

File: src/app/features/asset-list/asset-list.tsx [hunk 1]
```diff
@@ -78,15 +50,11 @@ export function BalancesList(): React.JSX.Element {
...
-      <CryptoCurrencyAssetItem
-        assetBalance={stxEffectiveBalance}
-        usdBalance={stxEffectiveUsdBalance}
-        address={address}
-        additionalBalanceInfo={stxAdditionalBalanceInfo}
-        additionalUsdBalanceInfo={stxAdditionalUsdBalanceInfo}
-        icon={<StxAvatar />}
-      />
-      <StacksFungibleTokenAssetList assetBalances={stacksFtAssetBalances} />
+
```
Comment (pete-watters) [inline src/app/features/asset-list/asset-list.tsx:53]: Does `{account => <StacksBalanceItem account={account} />}` mean to only pass and render `<StacksBalanceItem` is `account` ?

File: src/app/query/stacks/tokens/non-fungible-tokens/non-fungible-token-holdings.hooks.ts [hunk 2]
```diff
@@ -2,42 +2,19 @@ import { useMemo } from 'react';
...
-import {
-  useGetNonFungibleTokenHoldingsListQuery,
-  useGetNonFungibleTokenHoldingsQuery,
-} from './non-fungible-token-holdings.query';
+import { useGetNonFungibleTokenHoldingsListQuery } from './non-fungible-token-holdings.query';
 
-export function useAllAccountsNonFungibleTokenHoldingsTotal(accounts?: StacksAccount[]) {
-  const accountsNftHoldings = useGetNonFungibleTokenHoldingsListQuery(accounts);
+export function useAllAccountsNonFungibleTokenHoldingsTotal(accounts: StacksAccount[]) {
+  const accountsNftHoldings = useGetNonFungibleTokenHoldingsListQuery(accounts.map(a => a.address));
```
Comment (pete-watters) [inline src/app/query/stacks/tokens/non-fungible-tokens/non-fungible-token-holdings.hooks.ts]: We could map using `account` instead of `a`?

**Review by pete-watters — APPROVED** [review]
Nice work 👏

**Review by alter-eggo — APPROVED** [review]

File: src/app/features/asset-list/asset-list.tsx (hunk 1 above)
Comment (kyranjamie) [reply src/app/features/asset-list/asset-list.tsx:53]: The only purpose of this is that the component returns `null` if there's no account, so we can only render `<StacksBalanceItem account={account} />}` when it's defined

**Comment by kyranjamie** [comment]
Can be merged in isolation to other work

**Pull request merged**